gensim/test/test_fasttext.py

```diff
@@ -1300,6 +1306,253 @@ def _run(self, fin):
         self.assertTrue(np.allclose(_ARRAY, array))
 
 
+def _create_and_save_test_model(fname, model_params):
+    model = FT_gensim(**model_params)
+    lee_data = LineSentence(datapath('lee_background.cor'))
+    model.build_vocab(lee_data)
+    model.train(lee_data, total_examples=model.corpus_count, epochs=model.epochs)
+    gensim.models.fasttext.save_facebook_model(model, fname)
+    return model
+
+
+def calc_max_diff(v1, v2):
+    return np.max(np.abs(v1 - v2))
+
+
+class SaveFacebookFormatModelTest(unittest.TestCase):
+    """
+    This class containts tests that check the following scenario:
+
+    + create fastText model1
+    + save a model1 to Facebook binary file
+    + load the file as model2
+    + check if  model1 == model2
+    """
+
+    def _check_roundtrip_model_model(self, model_params):
+
+        with temporary_file("roundtrip_model_to_model.bin") as fpath:
+
+            model_trained = _create_and_save_test_model(fpath, model_params)
+
+            model_loaded = gensim.models.fasttext.load_facebook_model(fpath)
+
+            self.assertEqual(model_trained.vector_size, model_loaded.vector_size)
```

Comment:
These assertions belong outside of the context manager because they do not depend on the temporary file. The models are already loaded into memory, aren't they?

Please check your context manager blocks and ensure their scope is as tight as possible.

Reply:
Corrected.